### PR TITLE
Precommit fixes

### PR DESCRIPTION
**Pull request by saisankargochhayat**
I have added a manifest, removed `.coafile` and tried resolving most mypy errors.

### CODE REVIEW

File: kebechet/config.py [hunk 1]
```diff
@@ -222,37 +226,37 @@ def post(*args, **kwargs):
             kwargs.pop("verify", None)
             return original_post(*args, **kwargs, verify=verify)
 
-        requests.Session.post = post
+        requests.Session.post = post  # type: ignore
```
Comment (fridex) [inline kebechet/config.py:229]: Maybe we could drop these as we dropped igitt (see the comment above).

File: kebechet/managers/info/info.py [hunk 2]
```diff
@@ -35,19 +35,19 @@
 class InfoManager(ManagerBase):
     """Manager for submitting information about running Kebechet instance."""
 
-    def run(self) -> typing.Optional[dict]:
+    def run(self) -> typing.Optional[dict]:  # type: ignore
```
Comment (fridex) [inline kebechet/managers/info/info.py:38]: Why ignore?

File: kebechet/managers/version/version_test.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 """Tests for version manager."""
 
 import pytest
-import typing
+import typing  # noqa F401
```
Comment (fridex) [inline kebechet/managers/version/version_test.py:4]: Not used?

File: kebechet/config.py (hunk 1 above)
Comment (saisankargochhayat) [reply kebechet/config.py:229]: Can we remove `_tls_verification` method in config.py now that we don't use IGitt?

File: kebechet/managers/info/info.py (hunk 2 above)
Comment (saisankargochhayat) [reply kebechet/managers/info/info.py:38]: Seems like there is an incompatible type with the base class, any suggestion?
`kebechet/managers/info/info.py:38: error: Signature of "run" incompatible with supertype "ManagerBase"`

File: kebechet/managers/version/version_test.py (hunk 3 above)
Comment (saisankargochhayat) [reply kebechet/managers/version/version_test.py:4]: thanks, I have removed it now.

File: kebechet/managers/update/update.py [hunk 4]
```diff
@@ -336,15 +336,15 @@ def _should_update(self, package_name, new_package_version) -> tuple:
             return None, True
         elif len(response) == 1:
             response = list(response)[0]
-            commits = response.get_all_commits()
+            commits = response.get_all_commits()  # type: ignore
```
[the remark this answers is not in the document]
Comment (saisankargochhayat) [reply kebechet/managers/update/update.py:339]: Mypy doesn't seem to understand the underlying properties/methods from other packages.

File: kebechet/managers/update/update.py [hunk 5]
```diff
@@ -336,15 +336,15 @@ def _should_update(self, package_name, new_package_version) -> tuple:
...
             if len(commits) != 1:
                 _LOGGER.info(
                     "Update of package {package_name} to version {new_package_version} will not be issued,"
                     "the pull request as additional commits (by a maintaner?)"
                 )
                 return response, False
 
-            pr_number = response.id
+            pr_number = response.id  # type: ignore
```
[the remark this answers is not in the document]
Comment (saisankargochhayat) [reply kebechet/managers/update/update.py:347]: Same as above

File: kebechet/managers/update/update.py [hunk 6]
```diff
@@ -446,7 +446,7 @@ def _create_update(
                 ["Pipfile.lock"],
                 merge_request,
             )
-            return old_version, package_version, merge_request.id
+            return old_version, package_version, merge_request.id  # type: ignore
```
[the remark this answers is not in the document]
Comment (saisankargochhayat) [reply kebechet/managers/update/update.py:449]: `merge_request.id` is not recognized by mypy

File: kebechet/config.py [hunk 7]
```diff
@@ -294,7 +298,7 @@ def run_analysis(cls, analysis_id: str, origin: str, service: str) -> None:
                 analysis_id.startswith("provenance")
                 and manager_name == "thoth-provenance"
             ):
-                kebechet_manager = ThothProvenanceManager
+                kebechet_manager = ThothProvenanceManager  # type: ignore
```
[the remark this answers is not in the document]
Comment (saisankargochhayat) [reply kebechet/config.py:301]:
```
kebechet/config.py:296: error: Incompatible types in assignment (expression has type "Type[ThothAdviseManager]", variable has type "Union[ThothAdviseManager, ThothProvenanceManager]")
kebechet/config.py:302: error: Incompatible types in assignment (expression has type "Type[ThothProvenanceManager]", variable has type "Union[ThothAdviseManager, ThothProvenanceManager]")
kebechet/config.py:317: error: "ThothAdviseManager" not callable
kebechet/config.py:317: error: "ThothProvenanceManager" not callable
Found 6 errors in 2 files (checked 24 source files)
```

[the remark this answers is not in the document]
Comment (KPostOffice) [reply kebechet/config.py:301]: `Union[Type[ThothAdviseManager], Type[ProvenanceManager]]`?

File: kebechet/managers/update/update.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (KPostOffice) [reply kebechet/managers/update/update.py:449]: Use `Union[Tuple[str, str, Any], None]` for return type?

Here and below

File: kebechet/managers/info/info.py (hunk 2 above)
Comment (KPostOffice) [reply kebechet/managers/info/info.py:38]: Change `ManagerBase.run( )` and other managers to return `typing.Optional[typing.Dict[Any,Any]]` and see if that fixes the issue

File: kebechet/managers/update/update.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (KPostOffice) [reply kebechet/managers/update/update.py:339]: Use a type hint instead.  If you know what `response.get_all_commits()` can return you can do something like:
```
commits = response.get_all_commits()  # type: <T>
```

File: kebechet/managers/update/update.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (KPostOffice) [reply kebechet/managers/update/update.py:347]: Use a type hint instead. If you know what response.get_all_commits() can return you can do something like:
```
commits = response.get_all_commits()  # type: <T>
```

File: kebechet/managers/update/update.py [hunk 8]
```diff
@@ -520,10 +520,12 @@ def _create_initial_lock(
...
                 commit_msg, branch_name, body="", labels=labels
             )
-            _LOGGER.info(f"Initial dependency lock present in PR #{request.id}")
+            _LOGGER.info(
+                f"Initial dependency lock present in PR #{request.id}"  # type: ignore
+            )
         elif len(request) == 1:
             request = list(request)[0]
-            commits = request.get_all_commits()
+            commits = request.get_all_commits()  # type: ignore
```
Comment (KPostOffice) [inline kebechet/managers/update/update.py:528]: Please add a type hint

File: kebechet/services.py [hunk 9]
```diff
@@ -46,14 +46,14 @@ def _encode_fslash(slug: str):
...
-            "slug_method": _keep_slug.__func__,
+            "slug_method": _keep_slug.__func__,  # type: ignore
         },
         "gitlab": {
             "token": "GITLAB_KEBECHET_TOKEN",
             "service_type": ServiceType.GITLAB,
             "config_url": "https://gitlab.com/api/v4/projects/{slug}/repository/files/.thoth.yaml/raw?ref={branch}",
             "auth": {"header": "Private-Token", "value": "{token}"},
-            "slug_method": _encode_fslash.__func__,
+            "slug_method": _encode_fslash.__func__,  # type: ignore
```
Comment (KPostOffice) [inline kebechet/services.py:56]:
```
Callable[[str], str]
```
For both

File: kebechet/managers/info/info.py (hunk 2 above)
Comment (saisankargochhayat) [reply kebechet/managers/info/info.py:38]: Still seems to throw - `Signature of "run" incompatible with supertype "ManagerBase"`

File: kebechet/config.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (saisankargochhayat) [reply kebechet/config.py:301]: Thanks, that seems to work!

File: kebechet/managers/update/update.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (saisankargochhayat) [reply kebechet/managers/update/update.py:347]: Still doesn't seem to accept - `kebechet/managers/update/update.py:528: error: "Set[Any]" has no attribute "get_all_commits"
`

File: kebechet/services.py (hunk 9 above)
Comment (saisankargochhayat) [reply kebechet/services.py:56]: `"slug_method": _encode_fslash.__func__,  # type: Callable[[str], str]`
Just gives syntax error

**Comment by goern** [comment]
do we need a manifest for this repo?!

**Comment by saisankargochhayat** [comment]
> do we need a manifest for this repo?!

We use the python package anywhere, but just release purposes, we can have or remove the `check-manifest` and the manifest file. WDYT?
@KPostOffice @fridex

**Comment by KPostOffice** [comment]
> > do we need a manifest for this repo?!
> 
> We don't use the python package anywhere as far as I know, but just for release purposes, we can have or remove the `check-manifest` and the manifest file. WDYT?
> @KPostOffice @fridex
@saisankargochhayat 
If we upload source distributions of this to PyPI it's probably best to keep MANIFEST.in

**Comment by saisankargochhayat** [comment]
Should we merge this?

File: kebechet/config.py (hunk 1 above)
Comment (fridex) [reply kebechet/config.py:229]: Yes, I think so :+1:

**Pull request merged**